go/ethclient/wallet/wallet.go

```diff
@@ -46,3 +55,12 @@ func (m *inMemoryWallet) SignTransaction(chainID int, tx types.TxData) (*types.T
 func (m *inMemoryWallet) Address() common.Address {
 	return m.pubKeyAddr
 }
+
+func (m *inMemoryWallet) IncrementNonce() uint64 {
+	atomic.AddUint64(&m.nonce, 1)
+	return m.nonce - 1
```

Comment:
this should be: `return atomic.AddUint64(&m.nonce, 1) - 1`
If you do it in two steps, it becomes again vulnerable to races.

Reply:
Good point 👍